lib/rets/client.rb

```diff
@@ -81,39 +81,52 @@ def find(quantity, opts = {})
 
     def find_with_retries(opts = {})
       retries = 0
-      resolve = opts.delete(:resolve)
-      find_with_given_retry(retries, resolve, opts)
+      find_with_given_retry(retries, opts)
     end
 
-    def find_with_given_retry(retries, resolve, opts)
+    def find_with_given_retry(retries, opts)
       begin
-        find_every(opts, resolve)
+        find_every(opts)
       rescue NoRecordsFound => e
         if opts.fetch(:no_records_not_an_error, false)
           client_progress.no_records_found
           opts[:count] == COUNT.only ? 0 : []
         else
-          handle_find_failure(retries, resolve, opts, e)
+          handle_find_failure(retries, opts, e)
         end
       rescue AuthorizationFailure, InvalidRequest => e
-        handle_find_failure(retries, resolve, opts, e)
+        handle_find_failure(retries, opts, e)
       end
     end
 
-    def handle_find_failure(retries, resolve, opts, e)
+    def handle_find_failure(retries, opts, e)
       if retries < opts.fetch(:max_retries, 3)
         retries += 1
         client_progress.find_with_retries_failed_a_retry(e, retries)
         clean_setup
-        find_with_given_retry(retries, resolve, opts)
+        find_with_given_retry(retries, opts)
       else
         client_progress.find_with_retries_exceeded_retry_count(e)
         raise e
       end
     end
 
-    def find_every(opts, resolve)
-      params = {"QueryType" => "DMQL2", "Format" => "COMPACT"}.merge(fixup_keys(opts))
+    def find_every(opts)
+      params = {
+        "SearchType"          => opts[:search_type],
+        "Class"               => opts[:class],
```

Comment:
Went one step further and raised Argument Errors if missing either of those two params.